row_data_summary.py

```diff
@@ -0,0 +1,172 @@
+def s3_to_df(s3, filename):
+    response = s3.get_object(Bucket=BUCKET, Key=filename)
+    return pd.read_csv(response.get("Body"))
+
+
+def summarize_weekly(dfs):
+    """
+    Groups ROW counts by week and returns the count for each type
+    Parameters
+    ----------
+    dfs (list of dicts): Pandas dataframes stored in 'data' and metadata
+
+    Returns
+    ----------
+    output (dataframe): A dataframe where each row is one week and the
+                        columns are the totals for each type of ROW permit
+
+    """
+    weekly = []
+    for file in dfs:
+        df = file["data"]
+        # Some data from AMANDA we want to summarize by FOLDERTYPE (DS, EX, RW)
+        if "FOLDERTYPE" in df.columns:
+            df = pd.pivot_table(
+                df,
+                index=file["date_col"],
+                columns="FOLDERTYPE",
+                values=file["count_col"],
+                aggfunc="sum",
+            )
+            df = df.reset_index()
+        # Convert date column to datetime type
+        df[file["date_col"]] = pd.to_datetime(df[file["date_col"]])
+        # Grouping by week (starting on sunday)
+        week = df.resample("W-SAT", on=file["date_col"])[file["summary_cols"]].sum()
+        week = pd.DataFrame(week)
+        week["Measure"] = file["name"]
+        weekly.append(week)
+
+    # Combining dataframes
+    output = pd.concat(weekly)
+    output = output.reset_index()
+    output = pd.pivot_table(output, index="index", columns="Measure")
+    # Renaming columns to "FOLDERTYPE Measure"
+    output.columns = output.columns.map(" ".join)
+    output = output.fillna(0)
+
+    # Adding smartsheet data to AMANDA Totals
+    output["DS Applications Received"] = (
+        output["DS Applications Received"]
+        + output["Count Permits Commercial DS Permit Requests"]
+        + output["Count Permits Residential DS Permit Requests"]
+    )
+    # Drop smartsheet columns
+    output = output.drop(
+        columns=[
+            "Count Permits Commercial DS Permit Requests",
+            "Count Permits Residential DS Permit Requests",
+        ]
+    )
+    output = output.reset_index()
+    output["index"] = output["index"].dt.strftime("%Y-%m-%d")
+    output["index"] = output["index"] + "T00:00:00.000"
+    output = output.rename(
+        columns={
+            "Count Permits Extension Requests": "Extension Requests",
+            "index": "date",
+        }
+    )
+    return output
+
+
+def socrata_columns(df):
+    # Renames a dataframe's columns to align with what Socrata is expecting
+    df.columns = [c.lower() for c in df.columns]
+    df.columns = [c.replace(" ", "_") for c in df.columns]
+    return df
+
+
+def df_to_socrata(soda, df):
+    """
+    Replaces all the data in the socrata dataset with data in the dataframe.
+
+    Parameters
+    ----------
+    soda: sodapy client object
+    df : Pandas Dataframe
+
+    """
+    payload = df.to_dict("records")
+    soda.replace(DATASET, payload)
```

Comment:
Do you think any error checking is helpful here in case the open data portal API doesn't respond as expected?

This comment is tied to this particular function, but I think it may be helpful to consider more broadly. 

I do recognize that it may not be the most important thing, depending on how frequently this ETL will run and what degree of resilience we're targeting in this dataset.

Reply:
> what degree of resilience we're targeting in this dataset

This doesn't need to be indestructible. I currently just have it running daily and if we miss a few days it's not a big deal. This is aiming to replace a weekly email that Paloma sends out.

Reply:
Sounds good; thanks for the context on the reliability needs here.